Fix hreview backcompat

gRegorLove opened the pull request:
Fixes #155 

- rel="self bookmark" -> u-url
- class="reviewer" -> p-author h-card
- fix class="item hproduct", was backcompat parsing as vevent instead of hproduct
- adds unit tests
- comment out a unit test that was not making any assertions or failures.
- fix some syntax/warnings in newer versions of PHPUnit

artlung commented on tests/Mf2/ParserTest.php:
```diff
@@ -139,6 +139,8 @@ public function testParseEWithWhitespace() {
 	 * @group parseH
 	 */
 	public function testInvalidClassnamesContainingHAreIgnored() {
+		self::expectNotToPerformAssertions();
```
So is this a test that succeeds if no assertions are performed? I must be misunderstanding it.

artlung replied:
I'm more used to an assertion that a string is not present or not equal or not contains, but that works.

tantek reviewed (changes requested):
Should we switch from indiewebcamp.com references to indieweb.org and use https instead (including example.com URLs)?

gRegorLove commented:
Updated indieweb.org URL and https://example.com in tests.

Updated the test @artlung brought up so it has an assertion of no matches instead of calling fail() if a match is found.
